src/Language/PureScript/Errors.hs

```diff
@@ -1197,13 +1197,18 @@ renderBox = unlines
   dropWhileEnd p = reverse . dropWhile p . reverse
   whiteSpace = all isSpace
 
+toTypelevelString :: Type -> Maybe String
+toTypelevelString t = case evalTypelevelFunctions t of
+  (TypeLevelString str) -> Just str
+  _ -> Nothing
+
 evalTypelevelFunctions :: Type -> Type
 evalTypelevelFunctions = everywhereOnTypes go
   where
   go (TypeApp (TypeConstructor f) x)
     | f == primName "TypeString" = TypeLevelString $ unpack $ strip $ pack $ prettyPrintTypeAtom x
```

Comment:
Instead of packing and unpacking, I'd prefer an interpretation function which returns `Maybe Box`. That way, we can handle newlines properly, as well as multi-line types.